test/LibraryManager.Test/SemanticVersionComparisonTest.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Web.LibraryManager.Providers.Unpkg;
+
+namespace Microsoft.Web.LibraryManager.Test
+{
+    [TestClass]
+    public class SemanticVersionComparisonTest
+    {
+        [DataTestMethod]
+        [DataRow(null, "1.0.0", "1.0.0")]
+        [DataRow("1.0.0", null, "1.0.0")]
+        [DataRow(null, null, null)]
+        [DataRow("2.0.0", "1.10.1", "2.0.0")]
+        [DataRow("3.0.0", "3.0.0-beta1", "3.0.0")]
+        public void CompareSemanticVersion(string selfVersion, string otherVersion, string laterVersion)
+        {
+            List<SemanticVersion> versions = new List<SemanticVersion>();
+
+            versions.Add(SemanticVersion.Parse(selfVersion));
+            versions.Add(SemanticVersion.Parse(otherVersion));
+            versions.Sort();
+            
+            Assert.AreEqual(versions[1].OriginalText, laterVersion);
```

Comment:
>Assert.AreEqual(versions[1].OriginalText, laterVersion); [](start = 12, length = 56)

Sorry, I'm a bit dense this morning. What code of ours are we testing here?

Reply:
This is to test against two semantic versions and find out the latest version to verify it's as expected. (test the VersionCompletionEntry.compareSemanticVersion in a opposite way). The VersionCompletionEntry itself looks like not convenient to mock. Should I have this test?

---
In reply to: [205559622](https://github.com/aspnet/LibraryManager/pull/312#discussion_r205559622) [](ancestors = 205559622)

Reply:
Are we testing our code at all with this?

---
In reply to: [205568956](https://github.com/aspnet/LibraryManager/pull/312#discussion_r205568956) [](ancestors = 205568956,205559622)

Reply:
This can get removed. Instead, using an integration test would be better.

---
In reply to: [205584652](https://github.com/aspnet/LibraryManager/pull/312#discussion_r205584652) [](ancestors = 205584652,205568956,205559622)